lib/round-tracker.js

```diff
@@ -0,0 +1,78 @@
+import { createMeridianContract } from './meridian/contract.js'
+
+export async function createRoundGetter (pgClient) {
+  const contract = await createMeridianContract()
+
+  let sparkRound
+  const updateSparkRound = async (meridianRound) => {
+    sparkRound = await mapCurrentMeridianRoundToSparkRound({
+      meridianContractAddress: contract.address,
+      meridianRound: BigInt(meridianRound),
+      pgClient
+    })
+    console.log('SPARK round started: %s', sparkRound)
+  }
+
+  contract.on('RoundStart', updateSparkRound)
+  updateSparkRound(await contract.currentRoundIndex())
+
+  return () => sparkRound
+}
+
+export async function mapCurrentMeridianRoundToSparkRound ({
+  meridianContractAddress,
+  meridianRound,
+  pgClient
+}) {
+  let sparkRound
+
+  const { rows: [lastMeridianRecord] } = await pgClient.query(
+    'SELECT * FROM meridian_contract_versions ORDER BY last_spark_round DESC LIMIT 1'
+  )
+
+  // More events coming from the same meridian contract
+  if (lastMeridianRecord && lastMeridianRecord.contract_address === meridianContractAddress) {
+    sparkRound = BigInt(lastMeridianRecord.spark_round_offset) + meridianRound
+    await pgClient.query(
+      'UPDATE meridian_contract_versions SET last_spark_round = $1 WHERE contract_address = $2',
+      [sparkRound, meridianContractAddress]
+    )
+    console.log('Mapped %s IE round %s to SPARK round %s',
+      meridianContractAddress,
+      meridianRound,
+      sparkRound
+    )
+  } else {
+  // We are running for the first time and need to map the meridian round to spark round 1
+  // Or the contract address has changed
+    const lastSparkRound = BigInt(lastMeridianRecord?.last_spark_round ?? 0)
+    sparkRound = lastSparkRound + 1n
```

Comment:
awesome that you're using the native BigInt type 👏  I'm going to refactor the code I wrote to use this too

Reply:
Cool!

FWIW, Ethers v6 migrated to `BigInt`, it's a bummer we have to use v5 for now.

https://docs.ethers.org/v6/migrating/#migrate-bigint